### PR TITLE
Stability improvements and response queue

### PR DESCRIPTION
**Pull request by puddly**
These are some stability improvements and fixes to problems I'm encountering with my IKEA bulbs:

 - Sometimes `dataConfirm` can arrive before `dataRequest`'s response. The current codebase uses `waiter.match(obj)`, which will actually match and resolve *all* `dataRequest` response `waiter` objects, even if not all arrive. With a response queue, all waiters will match a single response, even if it arrived early.

   - Currently the queue will slowly fill up with unhandled responses. While the length is capped to 1,000 elements, this is not ideal. It would be more elegant to ignore responses that will never be used.

 - `dataRequest` responses can have a non-zero status, which should be considered a failure. This usually happens for me if the network is congested. For now I'm handling retrying at a higher level, but it might be useful to handle this within `zigpy-cc` itself (or not).

Right now there are no tests and I'm not very happy with my queue implementation so this is a draft pull request.

### CODE REVIEW

**Comment by sanyatuning** [comment]
Thanks for your work, I will test it sometime

**Comment by sanyatuning** [comment]
I added you write access to this repo, because I do not have enough time to actively develop it
I just want to start the work

**Comment by puddly** [comment]
@sanyatuning Thanks. I don't have a lot of time to devote to this project either but I will do what I can.

File: zigpy_cc/api.py [hunk 1]
```diff
@@ -97,17 +97,50 @@ def close(self):
...
+
+                        LOGGER.warning('Received a routing failure! Attempting to discover a new route before failing...')
+
+                        #await self.request(Subsystem.ZDO, 'extRouteDisc', {
+                        #    'dstAddr': obj.payload['dstaddr'],
+                        #    'options': 1,
+                        #    'radius': DEF_NWK_RADIUS
+                        #})
+
+                        await self.request(Subsystem.ZDO, 'extRouteDisc', {
```
Comment (Koenkk) [inline zigpy_cc/api.py]: AFAIK this command is only available on zStack 3.x.0 (3.0.x fails and 1.2 doesn't support it).

Comment (puddly) [reply zigpy_cc/api.py]: Thanks for the info. I'll take a look at the older versions as well.

Comment (Koenkk) [reply zigpy_cc/api.py]: I was wrong, it works on those version but make sure you use the following logic: https://github.com/Koenkk/zigbee-herdsman/blob/master/src/adapter/z-stack/adapter/zStackAdapter.ts#L191

File: zigpy_cc/types.py [hunk 2]
```diff
@@ -73,7 +74,7 @@ class ParameterType(uint8_t, enum.Enum):
...
     def __init__(self) -> None:
         self.networkKeyDistribute = False
-        self.networkKey = [1, 3, 5, 7, 9, 11, 13, 15, 0, 2, 4, 6, 8, 10, 12, 13]
-        self.panID = 0x1A62
-        self.extendedPanID = [0xDD, 0xDD, 0xDD, 0xDD, 0xDD, 0xDD, 0xDD, 0xDD]
-        self.channelList = [11]
+        self.networkKey = [0x6A, 0x8F, 0xBB, 0x75, 0xB2, 0xE8, 0xD9, 0x6D, 0xE5, 0xC5, 0x2D, 0x9D, 0x18, 0x56, 0x41, 0x1D]
+        self.panID = 0x8258
+        self.extendedPanID = [0x7A, 0xB2, 0x1E, 0x59, 0x19, 0x55, 0x46, 0x27]
+        self.channelList = [Common.LogicalChannel.CH15]
```
Comment (sanyatuning) [inline zigpy_cc/types.py]: Why did you change network options?

Comment (puddly) [reply zigpy_cc/types.py]: Even after a complete flash erase and re-flash, I could not form a network. Changing the PAN ID helped.

It would be better to have a way to configure these values directly from the controller application so that the codebase isn't polluted with these defaults, which I'll probably try doing in a bit.

Comment (puddly) [reply zigpy_cc/types.py]: A better question I think is why all of these settings are hard-coded to begin with. The bellows project picks a random pan ID and extended pan ID when forming a new network. I believe the security key is picked at random as well.

**Comment by sanyatuning** [comment]
Hi @puddly, why did you close this PR?

**Comment by puddly** [comment]
The zigbee2mqtt codebase doesn't translate well into idiomatic Python and I found myself struggling to figure out what was actually happening with all of the magic numbers and heavy passing around of dictionaries. Reorganizing and documenting everything would leave only the core zigbee2mqtt protocol definitions untouched, requiring double the work of a rewrite to end up with the same thing.

I'm going to focus [my efforts](https://github.com/puddly/zigpy-znp) on implementing the application layer from scratch based on [zha-new/zigpy-znp](https://github.com/zha-ng/zigpy-znp/).
